Vwitter/View Controllers/SelectAudienceViewController.m

```diff
@@ -0,0 +1,152 @@
+//
+//  SelectAudienceViewController.m
+//  Vwitter
+//
+//  Created by Christina Li on 7/8/22.
+//
+
+#import "SelectAudienceViewController.h"
+#import <Parse/Parse.h>
+#import "Follow.h"
+#import "AudienceMemberCell.h"
+#import "Vent.h"
+#import "VentAudience.h"
+#import <MBProgressHUD/MBProgressHUD.h>
+
+@interface SelectAudienceViewController () <UITableViewDataSource, UITableViewDelegate>
+@property (weak, nonatomic) IBOutlet UIButton *postVentButton;
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+@property (strong, nonatomic) NSMutableArray *arrayOfAudienceMembers;
+@property (strong, nonatomic) NSMutableArray *arrayOfSelectedAudience;
+
+@end
+
+@implementation SelectAudienceViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.arrayOfSelectedAudience = [[NSMutableArray alloc] init];
+    self.tableView.allowsMultipleSelection = YES;
+    [self loadData];
+
+    self.postVentButton.layer.cornerRadius = 20;
+    
+    self.tableView.dataSource = self;
+    self.tableView.delegate = self;
+    
+}
+
+- (void)loadData {
+    PFQuery *audienceQuery = [Follow query];
+    [audienceQuery whereKey:@"followingUserId" equalTo:[PFUser currentUser]];
+    [audienceQuery whereKey:@"approved" equalTo:@YES];
+    [audienceQuery includeKey:@"currentUserId"];
+    audienceQuery.limit = 20;
```

Comment:
Just like with the other stuff, I would refactor this to be a Parse Cloud function call. The problem is, this logic guarantees we can only show up to 20 people right? And so if you wanted to support showing more potential audience users, you would need to increase this number, but then we start getting really inefficient because the more data there is, the longer the request will take.

The better alternative will be to create some new parse cloud function, something like `"getPotentialAudienceUsers"`. In fact, even cooler, this query could easily support infinite scrolling, it could accept parameters like `page`, `maxResultCount`, and so on so that the client can load the users in batches of 20 for example, but as they scroll down, it can make a new request to load the next batch.

I found a good easy example online of how you could implement this "pagination" in Parse Cloud Code: https://stackoverflow.com/questions/20950648/parse-com-how-to-paginate-results-from-a-parse-query